core/generated-sources/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/ColumnsSelectionDsl.kt

```diff
@@ -0,0 +1,515 @@
+        }
+
+        df.select {
+            "name".first {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+
+        df.select {
+            Person::name.first {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+
+        df.select {
+            pathOf("name").first {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+    }
+
+    @Test
+    fun last() {
+        df.select { all().last() } shouldBe df.select { last() }
+        df.select { all().last() } shouldBe df.select { isHappy }
+        df.select { last() } shouldBe df.select { isHappy }
+        df.select { last { it.name().startsWith("a") } } shouldBe df.select { age }
+        df.select {
+            name.last {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select {
+            name.colsOf<String>().last {
+                it.any { it == "Alice" }
+            }
+        }
+
+        df.select {
+            "name".last {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+
+        df.select {
+            Person::name.last {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+
+        df.select {
+            pathOf("name").last {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+    }
+
+    @Test
+    fun single() {
+        val singleDf = df.select { take(1) }
+        singleDf.select { all().single() } shouldBe singleDf.select { single() }
+        singleDf.select { all().single() } shouldBe singleDf.select { name }
+        singleDf.select { single() } shouldBe singleDf.select { name }
+        df.select { single { it.name().startsWith("a") } } shouldBe df.select { age }
+
+        df.select {
+            name.single {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select {
+            name.colsOf<String>().single {
+                it.any { it == "Alice" }
+            }
+        }
+
+        df.select {
+            "name".single {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+
+        df.select {
+            Person::name.single {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+
+        df.select {
+            pathOf("name").single {
+                it.any { it == "Alice" }
+            }
+        } shouldBe df.select { name.firstName }
+    }
+
+    @Test
+    fun col() {
+        df.select { col("age") } shouldBe df.select { age }
+        df.select { col<Int>("age") } shouldBe df.select { age }
+        df.select { col(pathOf("age")) } shouldBe df.select { age }
+        df.select { col<Int>(pathOf("age")) } shouldBe df.select { age }
```

Comment:
yes, it definitely works like that. This is for completeness, since we also have `colGroup(path)` and `frameCol(path)`

Reply:
What if we make it "deprecated"? Instead of `df.select { col(pathOf("age")) }` use `df.select { pathOf("age") }`

Reply:
hmm, could do that... But then we'd need to do that for all identity functions. I'd rather just mention in the docs it doesn't do anything and leave it for completeness. Makes it also easy to refactor for users if they can just swap out `colGroup` for `col`.

Reply:
I will add a note in the docs for it though, cause I see it's missing

Reply:
the typed version is still useful though :)

Reply:
maybe it means we lack a typed version of pathOf? c: There should be "cast" though

Reply:
yeah you can also cast

Reply:
but I'd rather have one overload too many and know we've considered all cases than not enough because some obscure overload prevents it from working correctly